### PR TITLE
Properly check report pollers and report reachability flags

### PR DESCRIPTION
**Pull request by ShahabT**
## What changed?

Properly check report pollers and report reachability flags in enhanced DescribeTaskQueue.

## Why?

This is a bug.

## How did you test it?

Existing tests.

## Potential risks

none.

## Documentation

none.
## Is hotfix candidate?

yes.

### CODE REVIEW

**Review by Shivs11 — APPROVED** [review]
LGTM

**Review by yiminc — COMMENTED** [review]
Can we have some test for this?

File: service/matching/matching_engine.go [hunk 1]
```diff
@@ -1915,6 +1918,7 @@ func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pm *physicalTaskQueueM
 		metrics.NamespaceTag(pmImpl.ns.Name().String()),
 		metrics.TaskTypeTag(physicalTaskQueueParameters.taskType.String()),
 		metrics.PartitionTypeTag(physicalTaskQueueParameters.partitionType.String()),
+		metrics.VersionedTag(versioned),
```
Comment (yiminc) [inline service/matching/matching_engine.go:1921]: Is versioned coming from user provided value? Would metrics cardinality be a concern?

**Review by yiminc — APPROVED** [review]
Please add test coverage

File: service/matching/matching_engine.go (hunk 1 above)
Comment (ShahabT) [reply service/matching/matching_engine.go:1921]: no this can have only 3 values: "unversioned", "buildID", "versionSet"

**Pull request merged**